src/core/options-manager.ts

```diff
@@ -113,13 +113,22 @@ class OptionsManager {
     }
 
     const { value, type } = valueDescriptor;
-    if (type === ValueType.Complex && value instanceof Object) {
+    if (type === ValueType.Complex && value instanceof Object
+        && !(value instanceof Array)
+    ) {
```

Comment:
If `type === ValueType.Complex`, then it cannot be `Array`, isn't it? Do we need `!(value instanceof Array)` checking?